Source/Request.swift

```diff
@@ -414,3 +323,283 @@ extension Request: CustomDebugStringConvertible {
         return components.joined(separator: " \\\n\t")
     }
 }
+
+// MARK: -
+
+/// Specific type of `Request` that manages an underlying `URLSessionDataTask`.
+open class DataRequest: Request {
+
+    // MARK: Helper Types
+
+    enum Requestable: TaskConvertible {
```

Comment:
I'm not sure an `enum` with one case is the best here. I think a `struct` would be simpler. 
```swift
struct Requestable: TaskConvertible {
    let request: URLRequest
    
    func task(session: URLSession, adapter: RequestAdapter?, queue: DispatchQueue) -> URLSessionTask {
        var task: URLSessionTask!
        queue.sync { task = session.dataTask(with: request) }
        return task
    }
}
```